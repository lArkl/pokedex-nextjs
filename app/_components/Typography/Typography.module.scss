.xl {
  font-size: x-large;
}
.lg {
  font-size: larger;
}

.md {
  font-size: medium;
  font-weight: 600;
}

.sm {
  font-size: small;
}
