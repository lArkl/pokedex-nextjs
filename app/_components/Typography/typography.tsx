import classNames from 'classnames'
import { FC, ReactNode } from 'react'
import styles from './Typography.module.scss'

type TypographyVariant = 'sm' | 'md' | 'lg' | 'xl'

interface TypographyProps {
  children: ReactNode
  variant?: TypographyVariant
  className?: string
}

const typographyTypes: Record<
  TypographyVariant,
  { Component: FC<Omit<TypographyProps, 'variant'>>; className: string }
> = {
  xl: {
    Component: (props) => <h1 {...props} />,
    className: styles.xl,
  },
  lg: {
    Component: (props) => <h2 {...props} />,
    className: styles.lg,
  },
  md: {
    Component: (props) => <p {...props} />,
    className: styles.md,
  },
  sm: {
    Component: (props) => <p {...props} />,
    className: styles.sm,
  },
}

const Typography: FC<TypographyProps> = ({ children, variant = 'sm', className }) => {
  const { Component, className: typeClassName } = typographyTypes[variant]
  return <Component className={classNames([typeClassName, className])}>{children}</Component>
}

export default Typography
