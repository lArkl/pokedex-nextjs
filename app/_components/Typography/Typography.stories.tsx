import type { Meta, StoryObj } from '@storybook/react'

import Typography from './typography'

const meta = {
  title: 'Atoms/Typography',
  component: Typography,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {},
  args: {
    children: 'Text',
  },
} satisfies Meta<typeof Typography>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    variant: 'md',
  },
}

export const Small: Story = {
  args: {
    variant: 'sm',
  },
}

export const Large: Story = {
  args: {
    variant: 'lg',
  },
}
