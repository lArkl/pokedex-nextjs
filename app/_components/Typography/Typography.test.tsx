import { render, screen } from '@testing-library/react'
import Typography from './typography'

describe('Typography', () => {
  it('renders component', async () => {
    const { asFragment } = render(<Typography>text</Typography>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('shows small size', async () => {
    render(<Typography variant="sm">text</Typography>)
    const text = screen.getByText('text')
    expect(text).toBeInTheDocument()
  })
  it('shows medium size', async () => {
    render(<Typography variant="md">text</Typography>)
    const text = screen.getByText('text')
    expect(text).toBeInTheDocument()
  })
  it('shows large size as heading', async () => {
    render(<Typography variant="lg">text</Typography>)
    const text = screen.getByRole('heading')
    expect(text).toHaveTextContent('text')
  })

  it('accepts classname', async () => {
    render(<Typography className="class">text</Typography>)
    const text = screen.getByText('text')
    expect(text).toHaveClass('class')
  })
})
