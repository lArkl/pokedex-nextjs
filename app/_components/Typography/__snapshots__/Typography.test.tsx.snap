// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography > renders component 1`] = `
<DocumentFragment>
  <p
    class="_sm_bcfef7"
  >
    text
  </p>
</DocumentFragment>
`;
